### PR TITLE
[SPARK-6756] [MLLIB] add toSparse, toDense, numActives, numNonzeros, and compressed to Vector

### PR DESCRIPTION
**Pull request by mengxr**
Add `compressed` to `Vector` with some other methods: `numActives`, `numNonzeros`, `toSparse`, and `toDense`. @jkbradley

### CODE REVIEW

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala [hunk 1]
```diff
@@ -525,6 +558,34 @@ class DenseVector(val values: Array[Double]) extends Vector {
     }
     result
   }
+
+  override def numActives: Int = size
+
+  override def numNonzeros: Int = {
+    // same as values.count(_ != 0.0) but faster
+    var nnz = 0
+    values.foreach { v =>
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala]: Is foreach as fast as a while loop?

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala [hunk 2]
```diff
@@ -116,6 +116,39 @@ sealed trait Vector extends Serializable {
    *          with type `Double`.
    */
   private[spark] def foreachActive(f: (Int, Double) => Unit)
+
+  /**
+   * Number of active entries. Inactive entries are all zeros, while active entries could be zero.
```
Comment (jkbradley) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala]: Better phrasing: An "active entry" is an element which is explicitly stored, regardless of its value.  Note that inactive entries have value 0.

**Comment by jkbradley** [comment]
LGTM other than those small comments, and the MIMA issues

**Comment by jkbradley** [comment]
LGTM pending tests

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala (hunk 1 above)
Comment (mengxr) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala]: They should have similar performance.

**Comment by mengxr** [comment]
test this please

**Comment by mengxr** [comment]
Merged into master.

**Pull request closed without merging**